resource/js/vocab-search.js

```diff
@@ -7,18 +7,119 @@ const vocabSearch = Vue.createApp({
       languages: [],
       selectedLanguage: null,
       searchTerm: null,
-      autoCompeteResults: [],
-      languageStrings: null
+      searchCounter: null,
+      renderedResultsList: [],
+      languageStrings: null,
+      msgs: null
     }
   },
   mounted () {
     this.languages = SKOSMOS.languageOrder
     this.selectedLanguage = SKOSMOS.content_lang
-    this.languageStrings = SKOSMOS.language_strings[SKOSMOS.lang]
+    this.searchCounter = 0
+    this.languageStrings = SKOSMOS.language_strings[SKOSMOS.lang] ?? SKOSMOS.language_strings.en
+    this.msgs = SKOSMOS.msgs[SKOSMOS.lang] ?? SKOSMOS.msgs.en
+    this.renderedResultsList = []
+    document.addEventListener('click', this.onClickOutside)
+  },
+  beforeUnmount () {
+    document.removeEventListener('click', this.onClickOutside)
   },
   methods: {
     autoComplete () {
+      const delayMs = 300
+
+      // when new autocomplete is fired, empty the previous result
+      this.hideDropdown()
+
+      // cancel pending API calls when method is called
```

Comment:
I first misunderstood this comment and thought this would cancel the API calls "in the air". But it actually cancels just the timer. Maybe change to "cancel timer for upcoming API call" (note singular, as there can be only one), or something along those lines.